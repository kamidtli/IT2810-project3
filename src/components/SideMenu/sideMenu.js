import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Link } from 'react-router-dom';
import Drawer from '@material-ui/core/Drawer';
import Button from '@material-ui/core/Button';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import MenuIcon from '@material-ui/icons/Menu';

const useStyles = makeStyles((theme) => ({
  list: {
    width: 250,
  },
  fullList: {
    width: 'auto',
  },
  icon: {
    color: theme.palette.primary.contrastText,
  },
  link: {
    color: theme.palette.text.primary,
    textDecoration: 'none',
  },
}));

export default function SideMenu() {
  const classes = useStyles();
  const [state, setState] = React.useState({
    left: false,
  });

  const toggleDrawer = (side, open) => (event) => {
    if (
      event.type === 'keydown'
      && (event.key === 'Tab' || event.key === 'Shift')
    ) {
      return;
    }

    setState({ ...state, [side]: open });
  };

  const sideList = (side) => (
    <div
      className={classes.list}
      role="presentation"
      onClick={toggleDrawer(side, false)}
      onKeyDown={toggleDrawer(side, false)}
    >
      <List>
        <Link to="/" className={classes.link}>
          <ListItem button>
            <ListItemText primary="Home" />
          </ListItem>
        </Link>
        <Link to="/watchlist" className={classes.link}>
          <ListItem button>
            <ListItemText primary="Watchlist" />
          </ListItem>
        </Link>
      </List>
    </div>
  );

  return (
    <div>
      <Button onClick={toggleDrawer('left', true)} className={classes.icon}>
        <MenuIcon fontSize="large" />
      </Button>
      <Drawer
        anchor="left"
        open={state.left}
        onClose={toggleDrawer('left', false)}
      >
        {sideList('left')}
      </Drawer>
    </div>
  );
}
