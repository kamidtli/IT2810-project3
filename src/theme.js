import { createMuiTheme, responsiveFontSizes } from '@material-ui/core/styles';

const theme = createMuiTheme({
  palette: {
    primary: {
      main: '#F6AE2D',
    },
    secondary: {
      main: '#000d34',
    },
  },
  status: {
    danger: 'orange',
  },
});

const responsiveTheme = responsiveFontSizes(theme);

export default responsiveTheme;
