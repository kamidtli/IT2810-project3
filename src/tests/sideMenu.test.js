import React from 'react';
import { shallow } from 'enzyme';
import { ThemeProvider } from '@material-ui/styles';
import SideMenu from '../components/SideMenu/sideMenu';
import theme from '../theme';

describe('Side Menu component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(
      <ThemeProvider theme={theme}>
        <SideMenu />
      </ThemeProvider>
    );
    expect(wrapper.toJSON).toMatchSnapshot();
  });
});
