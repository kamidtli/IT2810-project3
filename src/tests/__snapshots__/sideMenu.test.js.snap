// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side Menu component renders without crashing 1`] = `undefined`;
